File: ruby/rna-transcription/rna_transcription.rb
class Complement
  DNA_TO_RNA_MAP = {
    'G' => 'C',
    'C' => 'G',
    'T' => 'A',
    'A' => 'U'
  }

  def self.of_dna(dna_strand)
    rna_strand = dna_strand.chars.map { |nucleotide|
      DNA_TO_RNA_MAP[nucleotide]
    }.join

    rna_strand.length == dna_strand.length ? rna_strand : ''
  end
end

module BookKeeping
  VERSION = 4
end

File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    dna_to_rna = {
        "G": "C",
        "C": "G",
        "T": "A",
        "A": "U"
    }

    rna = ""
    for nucleotide in dna_strand:
        if nucleotide in dna_to_rna:
            rna += dna_to_rna[nucleotide]
        else:
            return ""

    return rna
